### PR TITLE
Save lensing weight map in sys maps

### PR DESCRIPTION
**Pull request by joezuntz**
This saves the count of the number of objects in the lensing maps as a lensing weight map for later.
Update: now saves a summed weight, though this is the same is the count in our case.

### CODE REVIEW

File: txpipe/mapping/basic_maps.py [hunk 1]
```diff
@@ -123,17 +124,20 @@ def finalize(self, comm=None):
             mask[count_g1>0] = True
             mask[count_g2>0] = True
 
-            #  Don't think we want to save these at the moment
-            del count_g1
-            del count_g2
+            # We only need to keep one of these as they should be
+            # the same.
+            counts_g[b] = count_g2.reshape(self.pixel_scheme.shape).flatten()
```
Comment (EiffL) [inline txpipe/mapping/basic_maps.py]: Hum.... I'm slightly confused, why are g1 and g2 collected independently? Looks like it's possible to have bins with g1 contribution but no g2 from particular galaxies....

File: txpipe/sysmaps.py [hunk 2]
```diff
@@ -204,15 +204,18 @@ def run(self):
                 self.save_map(group, f"g2_{b}", map_pix, g2[b], config)
                 self.save_map(group, f"var_g1_{b}", map_pix, var_g1[b], config)
                 self.save_map(group, f"var_g2_{b}", map_pix, var_g2[b], config)
+                self.save_map(group, f"lensing_weight_{b}", map_pix, counts_g[b], config)
```
Comment (EiffL) [inline txpipe/sysmaps.py]: So.... I naively would have expected "lensing_weight" to mean inverse variance weight, but if I understand correclty this is more "number of objects per pixel", right?

**Comment by joezuntz** [comment]
We resolved to include a weight column and use that, even if it's 1 for metacal.

**Comment by joezuntz** [comment]
@EiffL I've updated this so that it now:
- generates a lensing weight column, all ones in the ingestion stage
- calculates a summed weight map per-pixel per-tomographic bin
- saves the weight maps (as before)

Let me know if this looks alright.

File: txpipe/mapping/basic_maps.py [hunk 3]
```diff
@@ -119,41 +131,42 @@ def finalize(self, comm=None):
...
-
-            v_g1 = v_g1.reshape(self.pixel_scheme.shape)
-            v_g2 = v_g2.reshape(self.pixel_scheme.shape)
-
-            g1[b] = mean_g1.flatten()
-            g2[b] = mean_g2.flatten()
+            # I really don't recall why I'm changing the shape
+            # back and forth here.  Maybe I had a reason?
+            # TODO: Remove the reshape/flatten and see what happens.
+            g1[b] = mean_g1.reshape(self.pixel_scheme.shape).flatten()
```
Comment (EiffL) [inline txpipe/mapping/basic_maps.py]: This looks harmless but yeah, I don't see the reason for the reshaping, I would just try to remove the reshapes before flattening.

File: txpipe/mapping/basic_maps.py (hunk 1 above)
Comment (joezuntz) [reply txpipe/mapping/basic_maps.py]: Added

**Comment by joezuntz** [comment]
I've dealt with the weird bits.  Many thanks for reviewing!

**Pull request merged**